src/ensembl/src/content/app/entity-viewer/state/sidebar/entityViewerSidebarActions.ts

```diff
@@ -93,6 +94,28 @@ export const updateEntityState = createAction(
   fragment: Partial<EntityViewerSidebarPayload>;
 }>();
 
+export const setSidebarInitialState = createAction(
+  'entity-viewer-sidebar/set-initial-state'
+)<string>();
+
+export const loadSidebar: ActionCreator<ThunkAction<
+  void,
+  any,
+  null,
+  Action<string>
+>> = () => (dispatch, getState: () => RootState) => {
+  const state = getState();
+  const genomeId = getEntityViewerActiveGenomeId(state);
+  const entityId = getEntityViewerActiveEnsObjectId(state);
+
+  if (!genomeId || !entityId) {
+    return;
+  }
+
+  dispatch(setSidebarInitialState(genomeId));
+  dispatch(fetchSidebarPayload());
+};
```

Comment:
I'm confused about what's happening to the sidebar in `loadSidebar` and `setSidebarInitialState`. What's the purpose of calling `setSidebarInitialState` in the `loadSidebar` here?

Reply:
I'm trying to ensure that the initial state for the `genomeId` is created before fetching the sidebar data. I guess the name `setSidebarInitialState` is misleading? 
I have renamed it to `setSidebarInitialStateForGenome`.

Reply:
> I'm trying to ensure that the initial state for the genomeId is created

Wouldn't it be better done in the reducer in response to active genome id change? It looks like a reducer responsibility.

We already have this by the way. Don't know if it's sufficient:
https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/content/app/entity-viewer/state/sidebar/entityViewerSidebarReducer.ts#L37-L43

Reply:
It is sufficient but I found it a bit confusing to use `generalActions` among the other actions here. 
The other problem I had was that `setActiveGenomeId` was not getting dispatched when we restore the state from the localStorage.

Reply:
> The other problem I had was that setActiveGenomeId was not getting dispatched when we restore the state from the localStorage.

I don't think we need the `setActiveGenomeId` at this point. Wouldn't we just add the stored parts of the state to sidebar's initial state here: 

https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/content/app/entity-viewer/state/sidebar/entityViewerSidebarState.ts#L99

Reply:
> It is sufficient but I found it a bit confusing to use generalActions among the other actions here.

I understand how it can be a bit surprising to see actions from one slice of the state influence the other slice, but I like this solution much more than creating an action for the same purpose that another action is already serving.

Reply:
> I don't think we need the `setActiveGenomeId` at this point. Wouldn't we just add the stored parts of the state to sidebar's initial state here:

If we don't dispatch `setActiveGenomeId`, the initial state for the genome will not be available when we fetch the sidebar data for the genome. This is why I thought it would be to have a separate action called `loadSidebar` to take care of setting the initial state as well as fetching the payload.

Reply:
`loadSidebar` is ok (although, after Imran's [PR](https://github.com/Ensembl/ensembl-client/pull/422), we might not need it any longer); but I would have expected the reducer to be able to figure out for itself what the initial state is when the app is bootstrapped.

Reply:
I guess is better to wait for Imran's PR to get merged first?

Reply:
Imran's PR was merged in. Meet the conflicts 😄

Reply:
Resolved the conflicts 👍